flocker/node/agents/functional/test_cinder.py

```diff
@@ -415,3 +419,117 @@ def test_disk_attachment_fails_with_conflicting_disk(self):
+        ``TimeoutException`` if ``/dev/disks/by-id/xxx`` does not appear within
+        5 seconds.
+        """
+        instance_id = self.blockdevice_api.compute_instance_id()
+        # Create volume
+        cinder_volume = self.cinder.volumes.create(
+            size=int(Byte(get_minimum_allocatable_size()).to_GiB().value)
+        )
+        self.addCleanup(self._cleanup, instance_id, cinder_volume)
+        volume = wait_for_volume_state(
+            volume_manager=self.cinder.volumes, expected_volume=cinder_volume,
+            desired_state=u'available', transient_states=(u'creating',))
+
+        # Suspend udevd before attaching the disk
+        [udev_process] = [
+            p for p in psutil.process_iter()
+            if p.name().endswith('-udevd')
+        ]
+        udev_process.suspend()
+        self.addCleanup(udev_process.resume)
+
+        # Attach volume
+        attached_volume = self.nova.volumes.create_server_volume(
+            server_id=instance_id,
+            volume_id=volume.id,
+            device=None,
+        )
+        volume = wait_for_volume_state(
+            volume_manager=self.cinder.volumes,
+            expected_volume=attached_volume,
+            desired_state=u'in-use',
+            transient_states=(u'attaching',),
+        )
+
+        e = self.assertRaises(
+            UnattachedVolume,
+            self.blockdevice_api.get_device_path,
+            volume.id,
+        )
+
+    @require_virsh
+    def test_getdevicepath_serial_number_matches_volume_id(self):
+        """
+        ``getdevicepath`` on systems using the virtioblk driver returns
+        a path matching ``/dev/disks/by-id/virtio-<volume.id>``.
+        """
+        instance_id = self.blockdevice_api.compute_instance_id()
+        # Create volume
+        cinder_volume = self.cinder.volumes.create(
+            size=int(Byte(get_minimum_allocatable_size()).to_GiB().value)
+        )
+        self.addCleanup(self._cleanup, instance_id, cinder_volume)
+        volume = wait_for_volume_state(
+            volume_manager=self.cinder.volumes, expected_volume=cinder_volume,
+            desired_state=u'available', transient_states=(u'creating',))
+
+        # Suspend udevd before attaching the disk
+        [udev_process] = [
+            p for p in psutil.process_iter()
+            if p.name().endswith('-udevd')
+        ]
+        udev_process.suspend()
+        self.addCleanup(udev_process.resume)
+
+        # Attach volume
+        attached_volume = self.nova.volumes.create_server_volume(
+            server_id=instance_id,
+            volume_id=volume.id,
+            device=None,
+        )
+        volume = wait_for_volume_state(
+            volume_manager=self.cinder.volumes,
+            expected_volume=attached_volume,
+            desired_state=u'in-use',
+            transient_states=(u'attaching',),
+        )
+        udev_process.resume()
+        sleep(5)
```

Comment:
Not clear why the udevd suspend and 5 second wait are required for this test. If it is required, please add more info to the docstring why we're testing this under these circumstances.